### ISSUE TITLE
When service 'http' is not defined, the service object gets rendered incorrectly

### ISSUE THREAD

**Issue by justSem** [bjw-s/helm-charts#127]
# Details

**What steps did you take and what happened:**

I've tried to use your template to deploy a factorio server - which doesn't have any http endpoints (or a service called 'http' for that matter')
I've found that when `service.main.ports.http` is not set, the template will render it in the service as an empty ports, and thus failing the validation done by kubernetes.

Helm debugging showed the service being rendered as this:
```yaml
# Source: app-template/templates/common.yaml
apiVersion: v1
kind: Service
metadata:
  name: factorio
  labels:
    app.kubernetes.io/service: factorio
    app.kubernetes.io/instance: factorio
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: factorio
    helm.sh/chart: app-template-1.3.2
  annotations:
spec:
  type: ClusterIP
  ports:
    - port: 34197
      targetPort: 34197
      protocol: UDP
      name: game
    - port:
      targetPort: http
      protocol: TCP
      name: http
    - port: 27015
      targetPort: 27015
      protocol: TCP
      name: steam
  selector:
    app.kubernetes.io/instance: factorio
    app.kubernetes.io/name: factorio
---
```

When setting `service.main.ports.http.enabled` explicitly to `false` the problem disappears.
However, when someone isn't aware of this mechanic kubernetes will only throw the 
`spec.ports[1].port: Invalid value: 0: must be between 1 and 65535, inclusive` making it very ambiguous to debug.

This issue happens because the template's [init file](https://github.com/bjw-s/helm-charts/blob/main/charts/library/common/templates/values/_init.tpl) makes a deepcopy reference to [values.yaml](https://github.com/bjw-s/helm-charts/blob/main/charts/library/common/values.yaml) which contains the defined `http` service with a non-defined port.

### To resolve the issue

I could dive deeper into this, haven't really done much with _writing_ helm templates before.
Basically my first guess would be to check whether the user has defined addtional services other then the default 'http' service and to omit the 'default' service if that's the case.

I hope this helps!

Kind regards,
Sem

**Comment by bjw-s** [comment]
I believe this is resolved in common-lib/app-template v2. Feel free to reopen the GitHub issue if that doesn't turn out to be the case.
